packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,106 @@
+# Select and Combobox option groups
+
+## Problem Statement
+
+It is a common request to categorize the options that are present in the dropdown for a `Select`/`Combobox` component. By supporting categories, developers are able to logically group options, and help end-users to quickly find the relevant option. This will also support use cases that need the same display values under separate categories.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Interaction Design](./IxD.md#groups)
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
```

Comment:
It'd be good to mention the plan for Angular and Blazor, even if it's just "no special considerations".

I scanned the [spec template](https://github.com/ni/nimble/blob/main/specs/templates/custom-component.md) and didn't see any other interesting missing sections besides a11y (mentioned below) but it couldn't hurt for you to review the template too in case anything else comes to mind.

Reply:
Done.